frontend/app/services/PayPalService.scala

```diff
@@ -1,32 +1,39 @@
 package services
 
+import actions.AuthRequest
+import com.gu.identity.play.IdMinimalUser
+import com.gu.paypal.PayPalConfig
+import com.gu.touchpoint.TouchpointBackendConfig.BackendType
 import com.netaporter.uri.Uri.parseQuery
 import com.typesafe.scalalogging.LazyLogging
 import configuration.Config
 import controllers.PayPal.{PayPalBillingDetails, Token}
 import controllers.routes
 import okhttp3.{FormBody, OkHttpClient, Request, Response}
-import play.api.mvc.RequestHeader
+import play.api.mvc.AnyContent
+import utils.TestUsers
 
 object PayPalService extends LazyLogging {
 
   // The parameters sent with every NVP request.
-  private val defaultNVPParams = Map(
-    "USER" -> Config.paypalUser,
-    "PWD" -> Config.paypalPassword,
-    "SIGNATURE" -> Config.paypalSignature,
-    "VERSION" -> Config.paypalNVPVersion)
+  private def defaultNVPParams(requestConfig : PayPalConfig) = Map(
+    "USER" -> requestConfig.user,
+    "PWD" -> requestConfig.password,
+    "SIGNATURE" -> requestConfig.signature,
+    "VERSION" -> requestConfig.NVPVersion)
 
   // Takes a series of parameters, send a request to PayPal, returns response.
-  private def nvpRequest(params: Map[String, String]) = {
+  private def nvpRequest(params: Map[String, String], user : IdMinimalUser) = {
+    logger.info("Is test user = " + TestUsers.isTestUser(user))
+    val requestConfig = TouchpointBackend.forUser(user).payPalConfig
 
     val client = new OkHttpClient()
```

Comment:
This looks like it'll create a new `OkHttpClient` with each request?

From the docs: 
https://github.com/square/okhttp/blob/71729c39f2/okhttp/src/main/java/okhttp3/OkHttpClient.java#L57

> OkHttpClients should be shared
> 
> OkHttp performs best when you create a single OkHttpClient instance and reuse it for all of your HTTP calls. This is because each client holds its own connection pool and thread pools. Reusing connections and threads reduces latency and saves memory. Conversely, creating a client for each request wastes resources on idle pools.

...I bet we break this rule elsewhere, but we definitely want to cut back here I think!